runelite-client/src/main/java/net/runelite/client/RuneLite.java

```diff
@@ -218,6 +223,31 @@ public void start(UpdateCheckMode updateMode) throws Exception
 
 		// Show UI after all plugins are loaded
 		clientUI.show();
+
+		if (isOutdated)
+		{
+			showReducedFunctionalityDialog();
+		}
+	}
+
+	private void showReducedFunctionalityDialog() throws Exception
```

Comment:
This should be in SwingUtil probably, we dont have anything Swing-specific in this class, except Applet stuff